### DISCUSSION TITLE
Suggestions for AttachedChar and stage interaction

### DISCUSSION THREAD

**Discussion by LasombraDemon** [ikemen-engine/Ikemen-GO#865] [Feature Request] [answered]
As we spoke in discord, some ideas:

-Stage constant redefinition via SCTRLs in AttachedChar would enable for actual stuff like Honda's breakable walls.

-Multiple stage platforms (impressive but sounds like a ton of work for the developers).

**_And something that might be even simpler:
-SCTRLs allowing to actually interact via AttachedChar with stage elements themselves (specially sprpriority, vel, and layerno)._**

**Comment by LasombraDemon** [comment]
Instead of opening a new suggestion thread, I thought I would add to this:

The only real limitiation I still run into, from AttachedChar, is layering. 

Stage elements are always *under* helpers and explods. Even with ModifyBG we can only go so far, since that can only touch already existing stage elements. Generating dynamic any AttachedChar elements means they will always be on top of stage elements.

I came up with a possible solution. A new parameter for explods and helpers, called "StageLayer".

All stages have their objects in ordered layers, correct? They are drawn as they are mentioned in the stage .def; this could place an object in _that_ specified layer order.

And this would not conflict with SprPriority, as this one is for the stage elements alone.

**Comment by potsmugen** [comment, marked as answer]
I think most of these have been implemented.
Modifying stage constants probably makes little sense, but I've thought about adding stage maps. But because stages don't run code in the same way chars do it feels a little awkward to have them (and an AC can just have its own maps already).
